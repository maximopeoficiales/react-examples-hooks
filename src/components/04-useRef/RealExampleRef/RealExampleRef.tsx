import { useState } from "react";
import MultipleCustomHooks from "../../03-examples/MultipleCustomHooks/MultipleCustomHooks";

interface MyProps {}
const defaultProps: MyProps = {};
const RealExampleRef = (props: MyProps) => {
  props = { ...defaultProps, ...props };

  const [hidden, setHidden] = useState(false);
  return (
    <div data-testid="RealExampleRef" className="row justify-content-center">
      <div className="col-md-10">
        <h1>RealExampleRef component</h1>
        <button
          onClick={() => {
            setHidden(!hidden);
          }}
        >
          Hidden
        </button>
        {hidden && <MultipleCustomHooks />}
      </div>
    </div>
  );
};

export default RealExampleRef;
