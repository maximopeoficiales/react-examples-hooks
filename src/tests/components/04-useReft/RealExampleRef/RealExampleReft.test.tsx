import { shallow } from "enzyme";
import React from "react";
import RealExampleRef from "../../../../components/04-useRef/RealExampleRef/RealExampleRef";

describe("Pruebas en RealExampleReft", () => {
  let wrapper = shallow(<RealExampleRef />);
  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBeFalsy();
  });

  test("debe mostrarse el componente <MultipleCustomHooks/>", () => {
    // const wrapper = shallow(<RealExampleRef />);
    let buttonHidden = wrapper.find("button");
    buttonHidden.simulate("click");
    console.log(wrapper.find("MultipleCustomHooks").html());
    expect(wrapper.find("MultipleCustomHooks").exists()).toBeTruthy();
  });
});
