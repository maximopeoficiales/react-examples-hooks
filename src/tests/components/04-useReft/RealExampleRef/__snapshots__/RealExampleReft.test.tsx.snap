// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RealExampleReft debe mostrarse correctamente 1`] = `
<div
  className="row justify-content-center"
  data-testid="RealExampleRef"
>
  <div
    className="col-md-10"
  >
    <h1>
      RealExampleRef component
    </h1>
    <button
      onClick={[Function]}
    >
      Hidden
    </button>
  </div>
</div>
`;
